akka-http/src/main/scala/akka/http/javadsl/server/JavaScalaTypeEquivalence.scala

```diff
@@ -0,0 +1,63 @@
+package akka.http.javadsl.server
+
+import akka.http.javadsl
+import akka.http.scaladsl
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+import java.util.Optional
+
+object JavaScalaTypeEquivalence {
+  // We define implicit conversions from the javadsl model objects to the scaladsl subclasses,
+  // for all types where there is only 1 valid direct subclass of the javadsl type
+  // (being the scaladsl one)
```

Comment:
As you noticed we have such concept in the form of `JavaMapping` so I'd like to keep those there too.
It's via implicits too there so we can simply provide an additional import there to enable conversions instead of asJava/asScala. I can help out with that for example.